llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -828,6 +841,11 @@ class VPRecipeBase : public ilist_node_with_parent<VPRecipeBase, VPBasicBlock>,
 
   /// Returns the debug location of the recipe.
   DebugLoc getDebugLoc() const { return DL; }
+
+protected:
+  /// Compute the cost of this recipe using the legacy cost model and the
+  /// underlying instructions.
+  InstructionCost computeCost(ElementCount VF, VPCostContext &Ctx) const;
```

Comment:
(Currently non-virtual, as only the base default implementation is provided, to be made virtual later.)